contracts/modules/TransactionManager.sol

```diff
@@ -177,33 +225,26 @@ abstract contract TransactionManager is BaseModule {
         require(_signature.length == 65, "TM: invalid signature length");
         address signer = Utils.recoverSigner(_msgHash, _signature, 0);
         require(_isOwner(msg.sender, signer), "TM: Invalid signer");
-        return ERC1271_ISVALIDSIGNATURE;
+        return ERC1271_IS_VALID_SIGNATURE;
     }
 
-    /**
-     * @notice Implementation of ERC721
-     * @notice An ERC721 smart contract calls this function on the recipient contract
-     * after a `safeTransfer`. If the recipient is a BaseWallet, the call to onERC721Received
-     * will be forwarded to the method onERC721Received of the present module.
-     * @return bytes4 `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
-     */
-    function onERC721Received(
-        address /* operator */,
-        address /* from */,
-        uint256 /* tokenId */,
-        bytes calldata /* data*/
-    )
-        external
-        returns (bytes4)
-    {
-        return ERC721_RECEIVED;
+
+    fallback() external {
+        bytes4 methodId = Utils.functionPrefix(msg.data);
+        if(methodId == ERC721_RECEIVED || methodId == ERC1155_RECEIVED || methodId == ERC1155_BATCH_RECEIVED) {
+            // solhint-disable-next-line no-inline-assembly
+            assembly {                
+                calldatacopy(0, 0, 0x04)
+                return (0, 0x20)
```

Comment:
The return value here is expected to be `bytes4` why return `bytes32`?

Reply:
I tried but it looks like in assembly you can't return less than 32 bytes.

Reply:
I don't think there's any limitation around assembly return. Doesn't `return (0, 0x04)` work?

Reply:
No :(

Reply:
Summary of discussion with Elena:
1) `return (0, 0x04)` works in `truffle` but makes `web3.eth.call()` return a `byte4` for static calls implemented via the module's fallback, which is inconsistent with the `byte32` normally returned by  `web3.eth.call()` when calling static methods returning a `byte4` but implemented _outside_ the fallback (i.e. with a proper method declaration).
2) `return (0, 0x04)` does not work in remix and yields `Failed to decode output: Error: data out-of-bounds (length=4, offset=32, code=BUFFER_OVERRUN, version=abi/5.0.7)` whereas `return (0, 0x20)` gives the exact same result as if the method was implemented with a proper method declaration.

Based on these observations, it seems to me that the EVM normally returns a "zero-padded `bytes32`" for when returning `bytes4`, even though libraries may reformat this "zero-padded `bytes32`" as a smaller type (e.g. bytes4) based on the available ABI.

In conclusion, to be safe, and since the goal of the fallback is to emulate the usage of a proper method declaration, I propose to use `return (0, 0x20)`.